srv/salt/_modules/osd.py

```diff
@@ -1793,8 +1793,7 @@ def partitions(self, osd_id):
 
         for device_type in ['journal', 'block', 'block.db', 'block.wal', 'block_dmcrypt']:
             result = self._uuid_device("{}/{}".format(mount_dir, device_type))
-            if result:
-                _partitions[device_type] = result
+            _partitions[device_type] = result
```

Comment:
the `if result:` is still needed as the `partitions` method also feeds the grains.